### PR TITLE
🏗 🐛 Fix bento peer dependency remapping on Windows

### PR DESCRIPTION
**Pull request by rileyajones**
When attempting to fix Bento component peer remapping, I introduced a bug where the remapped paths do not work on Windows.

#37797

### CODE REVIEW

File: build-system/common/esbuild-babel.js [hunk 1]
```diff
@@ -127,7 +127,7 @@ function getEsbuildBabelPlugin(
         const map = outputFiles.find(({path}) => path.endsWith('.map'));
 
         if (!map) {
-          debug('post-esbuild', code.path, code.text);
+          debug('post-esbuild', code?.path, code?.text);
```
Comment (jridgewell) [inline build-system/common/esbuild-babel.js:130]: Unrelated changes?

**Review by jridgewell — APPROVED** [review]

File: build-system/common/esbuild-babel.js (hunk 1 above)
Comment (rileyajones) [reply build-system/common/esbuild-babel.js:130]: This got merged due to auto-merge being enabled.
This is an unrelated change but I have seen situations where code is `undefined` due to some build error that this then masks. Due to that I would rather leave this in place then revert but I'm fine with that as well.
